edgelet/hsm-sys/azure-iot-hsm-c/src/edge_pki_openssl.c

```diff
@@ -105,20 +105,113 @@ extern time_t get_utc_time_from_asn_string(const unsigned char *time_value, size
 //#################################################################################################
 // PKI key operations
 //#################################################################################################
+
+// implementation from https://wiki.openssl.org/index.php/EVP_Signing_and_Verifying
+static int caluclate_hmac_sha256
+(
+    EVP_PKEY *evp_key,
+    const unsigned char *tbs,

```

Comment:
Who sets the "to be signed"?

Reply:
This is the buffer passed in by the caller. It is const as it only needs be read. See cert_key_sign below.